From db2ac8a4a2a8938db7e41f36c82ba50dcbd2bf4f Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 17 Aug 2018 16:11:32 +0800
Subject: [PATCH 4530/5242] MLK-19413-19 gpu: imx: dpu: framegen: Do not warn
 on zero pclk rate for slave framegen

commit  379f5b1439fb4a1ae46c0232d6da058c6c267b55 from
https://source.codeaurora.org/external/imx/linux-imx.git

The pixel clock of the slave framegen comes from the master framegen
when the two framegens work in sync mode, so we cannot get pixel clock
from the pixel clock rate from the slave framegen directly to calculate
the timeout value when waiting for it's done.  Instead, we can tell
the rate from the video mode.  So, we take this fallback way to get
the rate in this case and don't warn.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-framegen.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dpu/dpu-framegen.c b/drivers/gpu/imx/dpu/dpu-framegen.c
index a52a264..54ff9d6 100644
--- a/drivers/gpu/imx/dpu/dpu-framegen.c
+++ b/drivers/gpu/imx/dpu/dpu-framegen.c
@@ -480,7 +480,8 @@ void framegen_wait_done(struct dpu_framegen *fg, struct drm_display_mode *m)
 		/* fall back to display mode's clock */
 		dotclock = m->crtc_clock;
 
-		dev_warn(fg->dpu->dev,
+		if (!(fg->side_by_side && framegen_is_slave(fg)))
+			dev_warn(fg->dpu->dev,
 				"pixel clock for FrameGen%d is zero\n", fg->id);
 	}
 
-- 
1.7.9.5

